com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -333,6 +333,9 @@ public IReadOnlyList<ulong> ConnectedClientsIds
         /// </summary>
         public bool IsConnectedClient { get; internal set; }
 
+
+        public bool ShutdownInProgress { get; internal set; }
```

Comment:
Really, we should have a few states for the NetworkManager to provide the NetworkManager's current state.
This is a minimal fix in order to determine if we are in the middle of a shutdown.

Reply:
instead of introducing this semi-fix/band-aid, what if we were to make `NetworkManager.Singleton = null` **after** we call `NetworkManager.Shutdown` (therefore destroy NOs first then set it to null)?

Reply:
That would work when there is a single instance running, but our MultiInstsance tests would fail.

Reply:
huh? why?

Reply:
I'm not suggesting to force `NetworkManager.Singleton = null`, I'm saying if we call network destroy after something sets `NetworkManager.Singleton = null` then let's call network destroy on objects first and later let that other code do `NetworkManager.Singleton = null`

Reply:
Well, either case it appear we have another [bool that signals the shutdown process that popped up recently](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/commit/5209fdca310fd7229f610d34bcdca3232f1965d2).
For now, I will just make it this: 
` public bool ShutdownInProgress { get { return m_ShuttingDown; } }`

I could make it internal but then users wouldn't be able to check this value for any edge case scenarios, which I could do this either way.